sklearn/linear_model/_theil_sen.py

```diff
@@ -375,15 +377,14 @@ def _check_subparams(self, n_samples, n_features):
         else:
             n_subsamples = min(n_dim, n_samples)
 
-        if self.max_subpopulation <= 0:
-            raise ValueError(
-                "Subpopulation must be strictly positive ({0} <= 0).".format(
-                    self.max_subpopulation
-                )
-            )
-
+        self._max_subpopulation = check_scalar(
+            self.max_subpopulation,
+            "max_subpopulation",
+            target_type=numbers.Integral,
```

Comment:
This is technically a breaking change: we used to accept floating point values and automatically cast those as int and now we will raise an error when upgrading scikit-learn for users that have code such as `TheilSenRegressor(max_population=1e3)`.

I think we should accept any real value for backward compat reasons:


```suggestion
             # should be numbers.Integral but can accept float for backward compat
             # reasons
            target_type=numbers.Real,
```

Reply:
the rest of the tests and docstring will need an update.

Note that the current docstring reads:

```
max_subpopulation : int, default=1e4
```